### ISSUE TITLE
Suggestion: Scroll to adjust screen brightness

### ISSUE THREAD

**Issue by adamp4** [RamonUnch/AltDrag#178]
Hi Ramon,
I'm a user of Altdrag and found your introduction from https://github.com/stefansundin/altdrag/discussions/128, just tried and then love it, very impressive improvements & optimization. The SmartAero fixed gap between windows which occured in Altdrag, and the translation files are seperated in folder which is convenient to customize.
Thanks a lot to you and Stefan

Best wishes
Adam Wu

**Comment by RamonUnch** [comment]
Good idea for a wheel action, I will see on how to implement this properly

**Comment by RamonUnch** [comment]
> translation files are seperated in folder which is convenient to customize.

If you have any improvements to make to some translation feel free to contribute here: #35, or submit a PR.

**Comment by adamp4** [comment]
Hi Ramon,
Any progress of this suggestion? Sorry that i really don't wanna bother you, all the devlopment decision should be decided by yourself, just thought you might fogot...
Best wishes 
Adam

> > translation files are seperated in folder which is convenient to customize.
> 
> If you have any improvements to make to some translation feel free to contribute here: #35, or submit a PR.

**Comment by ZhichaoHong** [comment]
Among server tools and monitor HW buttons, this software 
https://github.com/emoacht/Monitorian works the best for me.

**Comment by adamp4** [comment]
> Among several tools and monitor HW buttons, this software https://github.com/emoacht/Monitorian works the best for me.

Thanks for ur suggestion, it works fine, but i already knew similar apps like [https://github.com/blackholeearth/Win10_BrightnessSlider], or better to use command/shortcut : nircmd.exe changebrightness 1 3.
The reason i suggested is mouse wheel seems more convenient and will make Altsnap better^^

**Comment by RamonUnch** [comment]
I have not worked on this much because I do not have software control to change brightness on my monitor (old monitor), so I cannot directly test the feature. I did write something, you can give it a try.

This is a logging version that will display brightness info in the ad.log file.
To use you must manually set `Scroll=Brightness` in the [Input] section of your ini file, using the test version below.

[AltSnap1.62test17_x64.zip](https://github.com/RamonUnch/AltSnap/files/13402810/AltSnap1.62test17_x64.zip)
It is untested, It might even crash, so be careful and report if it works for you, also post the content of ad.log.

**Comment by adamp4** [comment]
> I have not worked on this much because I do not have software control to change brightness on my monitor (old monitor), so I cannot directly test the feature. I did write something, you can give it a try.
> 
> This is a logging version that will display brightness info in the ad.log file. To use you must manually set `Scroll=Brightness` in the [Input] section of your ini file, using the test version below.
> 
> [AltSnap1.62test17_x64.zip](https://github.com/RamonUnch/AltSnap/files/13402810/AltSnap1.62test17_x64.zip) It is untested, It might even crash, so be careful and report if it works for you, also post the content of ad.log.
Thanks Ramon, it doesn't work.
[ad.log](https://github.com/RamonUnch/AltSnap/files/13402872/ad.log)

**Comment by RamonUnch** [comment]
Try again with this hooks.dll (more info to be printed)

[hooks1.62test18_x64.zip](https://github.com/RamonUnch/AltSnap/files/13402949/hooks1.62test18_x64.zip)

**Comment by adamp4** [comment]
> Try again with this hooks.dll (more info to be printed)
> 
> [hooks1.62test18_x64.zip](https://github.com/RamonUnch/AltSnap/files/13402949/hooks1.62test18_x64.zip)

[ad.log](https://github.com/RamonUnch/AltSnap/files/13403002/ad.log)

**Comment by RamonUnch** [comment]
Strange, maybe I am using the wrong API, try again with this one.
delete your ad.log file before testing so I do not have to scroll down too much.
[hooks1.62test19_x64.zip](https://github.com/RamonUnch/AltSnap/files/13403651/hooks1.62test19_x64.zip)

**Comment by RamonUnch** [comment]
Are you using a laptop? According to some sources with laptops we should use `WmiMonitorBrightnessMethods` interface instead of using the usual MCCS/I2C interface.

**Comment by adamp4** [comment]
Yes, i'm using a surface pro 3.
[ad.log](https://github.com/RamonUnch/AltSnap/files/13404743/ad.log)

**Comment by RamonUnch** [comment]
I will investigate using the other API then, from the log it seems clear that the protocol is not handled by your monitor.
I will inform you of course.
